r/R/dataset-write.R

```diff
@@ -60,8 +62,13 @@ write_dataset <- function(dataset,
                           format = c("parquet", "feather", "arrow", "ipc"),
                           partitioning = dplyr::group_vars(dataset),
                           basename_template = paste0("part-{i}.", as.character(format)),
-                          hive_style = TRUE,
+                          hive_style = TRUE, max_partitions = 1024L,
                           ...) {
+  stopifnot(
+    max_partitions == round(max_partitions, 0),
+    max_partitions == abs(max_partitions),
+    !is.null(max_partitions)
+  )
```

Comment:
Have you tried to leave this checking off and seen what errors the c++ code returns? If those errors are reasonable, we should use them instead of writing our own here.

Reply:
yes, I wrote that because I can pass max_partitions = -3
and it's gonna create 3 partitions, which is very confusing
also, NA works, and goes to the default 1024L, but NULL fails

Reply:
Can you tell if that's a feature or a bug in c++? This partitions parameter is pretty low-level so I think we should stick with what c++ allows (and fix this if it's a bug in c++, though I'm not certain it is)

Reply:
related https://issues.apache.org/jira/browse/ARROW-12373